### ISSUE TITLE
Allowing rule ids with "-" breaks (non standard) additional texts

### ISSUE THREAD

**Issue by offa** [priv-kweihmann/oelint-adv#679]
I add some reason text to `nooelint` suppression for documentation:

```python
# nooelint: oelint.vars.insaneskip - some good reason
INSANE_SKIP:${PN} += "already-stripped"
```

This worked until 6.6.7 where `-` became allowed for ids  (#678), so the suppression of the example above wont work anymore:

```
oelint.file.inlinesuppress_na:Inline suppression for 'oelint.vars.insaneskip - some good reason' is not needed
oelint.vars.insaneskip:INSANE_SKIP should be avoided at any cost 
```
The full text is considered for the id, even though `-` is surrounded by spaces.

I don't consider this as a bug, since there's no support for these kind of text, but is there are recommended way to add some kind of justification to the suppression in 6.6.7+?

**Comment by priv-kweihmann** [comment]
Sadly that slipped through on the last bug fix, even if undocumented that ofc is a very welcome feature.
Fix is pending.

But it comes with the expense that unknown IDs like foo.bar.baz are now comments, which means if they are not linked to
any of the currently configured rules, it will not generate a warning - to me that is compromise
